### PR TITLE
CraftComponent Entity + Dao

### PR DESCRIPTION
**Pull request by JanCizmar**

### CODE REVIEW

File: spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java [hunk 1]
```diff
@@ -1,12 +1,75 @@
...
+import javax.validation.constraints.NotNull;
 import java.time.ZonedDateTime;
 
+@Entity
 public class CraftComponent {
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Long id;
 
+	@Column
```
Comment (mlynarikj) [inline spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java]: column not necessary same as notNull annotation as this is not a nullable object

File: spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java [hunk 2]
```diff
@@ -1,12 +1,75 @@
...
+	@NotNull
 	private boolean readyToUse;
+
+	@Column
```
Comment (mlynarikj) [inline spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java]: add unique to column

File: spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java [hunk 3]
```diff
@@ -1,12 +1,75 @@
...
+		return readyDate;
+	}
+
+	public void setReadyDate(ZonedDateTime readyDate) {
+		this.readyDate = readyDate;
+	}
+
+
+	@Override
+	public boolean equals(Object o) {
```
Comment (mlynarikj) [inline spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java]: make hash and equals dependant only on name

File: spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java [hunk 4]
```diff
@@ -1,12 +1,75 @@
...
+		if (this == o) return true;
+		if (o == null || getClass() != o.getClass()) return false;
+
+		CraftComponent that = (CraftComponent) o;
+
+		if (isReadyToUse() != that.isReadyToUse()) return false;
```
Comment (Broseten) [inline spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java]: I think that "primary key" (or the set that defines equality) does not need to include this bool, what do others think?
What defines unique component?

**Review by mlynarikj — CHANGES_REQUESTED** [review]
only some small changes

File: spaceMissionManager-persistance/src/main/java/cz/muni/fi/Dao/CraftComponentDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Override
+    public void removeComponent(CraftComponent craftComponent) {
+        if(craftComponent == null){
+            throw new IllegalArgumentException("craftComponent is null");
+        }
+        if (craftComponent.getId() == null) {
+            throw new IllegalArgumentException("id is null");
+        }
+        entityManager.remove(this.entityManager.merge(craftComponent));
```
Comment (Broseten) [inline spaceMissionManager-persistance/src/main/java/cz/muni/fi/Dao/CraftComponentDaoImpl.java:74]: Why using merge not find?

File: spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java [hunk 6]
```diff
@@ -1,12 +1,75 @@
...
+	private Long id;
 
+	@Column
+	@NotNull
 	private boolean readyToUse;
+
+	@Column
+	@NotNull
 	private String name;
+
```
Comment (Broseten) [inline spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java]: Missing Spacecraft realtionship?

File: spaceMissionManager-persistance/src/main/java/cz/muni/fi/Dao/CraftComponentDaoImpl.java (hunk 5 above)
Comment (mlynarikj) [reply spaceMissionManager-persistance/src/main/java/cz/muni/fi/Dao/CraftComponentDaoImpl.java:74]: both options are possible as far as I can tell and shouldn't be that much different

File: spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java (hunk 6 above)
Comment (jsmadis) [reply spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java]: Don't forget to add set and get for Spacecraft

File: spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java (hunk 2 above)
Comment (JanCizmar) [reply spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java]: What if we need to add same component to other spacecraft? Should the name be really unique?

File: spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java [hunk 7]
```diff
@@ -1,12 +1,77 @@
...
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Long id;
 
 	private boolean readyToUse;
+
+	@Column(unique = true)
+	@NotNull
 	private String name;
+
+	private Spacecraft spacecraft;
```
Comment (mlynarikj) [inline spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/CraftComponent.java]: add @ManyToOne(fetch = FetchType.LAZY) annotation

**Review by mlynarikj — CHANGES_REQUESTED** [review]

**Review by mlynarikj — APPROVED** [review]

**Pull request merged**